### ISSUE TITLE
Updated typing breaks Trans componont typing

### ISSUE THREAD

**Issue by razzeee** [i18next/react-i18next#1655]
## 🐛 Bug Report

After updating to `@types/react` v18, the `<Trans>` component complains, when used with `{{value}}` placeholders.

It's obvious from https://github.com/flathub/website/actions/runs/5349795927/jobs/9701520404?pr=843 and I was asked to do a minimal reproduction and open this https://github.com/i18next/react-i18next/issues/1547#issuecomment-1603262601

## To Reproduce

Check out this repo https://github.com/razzeee/i18next-react-bug

Install the dependencies and you should be able to see the type error in `index.tsx`

## Expected behavior

The typing should be able to handle strings in these places I think?

## Your Environment

Shouldn't matter and be encapsulated in the minimal repro

**Comment by adrai** [comment]
The link to your repo seems broken @razzeee

**Comment by razzeee** [comment]
Sorry, tried to use vscode to create the repo and probably misclicked. Should be better now.

**Comment by adrai** [comment]
Interesting, removing the `node_modules/i18next/index.d.ts` file indeed removes that ts error...
I have no idea why this happens...
@pedrodurek do you know why?

**Comment by adrai** [comment]
Also uncommenting this line removes that ts error:

<img width="386" alt="image" src="https://github.com/i18next/react-i18next/assets/1086194/f59bcf1c-cb66-4904-85b6-debd630b2a81">

**Comment by adrai** [comment]
ahhh... the error also is removed if you set allowObjectInHTMLChildren to true

**Comment by adrai** [comment]
fyi: https://github.com/i18next/react-i18next/issues/1543

**Comment by adrai** [comment]
> ahhh... the error also is removed if you set allowObjectInHTMLChildren to true Can you confirm @razzeee ?
> 
> tldr; without i18next you get the same error: <img alt="image" width="719" src="https://user-images.githubusercontent.com/1086194/254492221-b2d0fb6c-b00b-43be-9752-47b4f7f7ce68.png">

@razzeee If you remove the react-i18next and i18next dependecies from the project, you get the same error also with normal <span> elements... without Trans component...

**Comment by razzeee** [comment]
I will need to wait for next week to check this

**Comment by razzeee** [comment]
> allowObjectInHTMLChildren

It seems to work in some cases.

In others I'm now getting 

```
Type '{ login_name: string; }' is not assignable to type 'ReactNode'.
  Object literal may only specify known properties, and 'login_name' does not exist in type 'ReactElement<any, string | JSXElementConstructor<any>> | ReactFragment | ReactPortal'.ts(2322)
```

Which might be what you where hinting at above?

**Comment by pedrodurek** [comment]
Hey @razzeee, can you share an example where it doesn't work?

**Comment by razzeee** [comment]
Hey, @pedrodurek  I'm not sure how to create a minimal example with that. But our pipeline here fails https://github.com/flathub/website/pull/843

https://github.com/flathub/website/actions/runs/5647791745/job/15298643966?pr=843

**Comment by adrai** [comment]
It's the same issue as the comment I've written before.
If you remove all i18next dependencies and weite the same <span> element it will show the same error... so not really related to i18next?

**Comment by razzeee** [comment]
@adrai thanks, I went digging again and figured out that the new types somehow don't like the `@` in those places. So it's indeed not related to i18next.
